Feature: Line to Arachne

azurejelly opened the pull request:
## What
This PR adds a setting called "Line to Arachne", which, just like the "Line to Miniboss" setting specific to slayers, draws a line to where Arachne is currently at. It requires the player to be in a 10 block range.

<details>
<summary>Images</summary>

![2024-05-24_23 06 15](https://github.com/hannibal002/SkyHanni/assets/156628929/d612b7f6-62f9-4715-856a-59d3c712b3e8)
![2024-05-24_23 06 23](https://github.com/hannibal002/SkyHanni/assets/156628929/59024c73-f7a0-41a8-af8f-d5fcfee90d17)
![2024-05-24_23 06 31](https://github.com/hannibal002/SkyHanni/assets/156628929/fa26e745-710c-4da2-92e0-99c8cba360ef)

</details>

## Changelog New Features
+ Added a "Line to Arachne" setting, just like with slayer minibosses. - azurejelly

Commit:
Feature: Draw line to Arachne, like with slayer minibosses

CalMWolfs commented on src/main/java/at/hannibal2/skyhanni/config/features/combat/MobsConfig.java:
```diff
@@ -57,6 +57,12 @@ public class MobsConfig {
+    @ConfigOption(name = "Line to Arachne", desc = "Also draw a line pointing to where Arachne is currently at.")
+    @ConfigEditorBoolean
+    @FeatureToggle
+    public boolean lineToArachne = true;
```
Should probably be default off

CalMWolfs commented on src/main/java/at/hannibal2/skyhanni/features/combat/mobs/MobHighlight.kt:
```diff
@@ -122,5 +158,7 @@ class MobHighlight {
         RenderLivingEntityHelper.setEntityColorWithNoHurtTime(entity, LorenzColor.RED.toColor().withAlpha(50))
         { config.arachneBossHighlighter }
+
+        this.arachne = entity
```
This code will only run if the user has arachneBossHighlighter on

CalMWolfs reviewed (changes requested):
Currently this feature requires another feature to be on to work

Commit:
Fix: Disable "Line to Arachne" by default

Commit:
Fix: Do not require the Arachne Boss Highlighter to be enabled in order to use Line to Arachne

Commit:
Fix: Update "Line to Arachne" description

It no longer requires "Arachne Boss" to be enabled, so "Also" in this context doesn't make much sense

CalMWolfs reviewed (approved):
lgtm

hannibal002 committed:
Merge branch 'refs/heads/beta' into fork/feature/line-to-arachne

hannibal002 committed:
code cleanup
